pkg/datamover/testutils.go

```diff
@@ -0,0 +1,438 @@
+		},
+	}
+	nsCreated, err := cli.CoreV1().Namespaces().Create(ctx, ns, metav1.CreateOptions{})
+	if err != nil {
+		return nil, err
+	}
+	return nsCreated, nil
+}
+
+func getKanisterToolsImage() string {
+	if val, ok := os.LookupEnv(kanisterToolsImageEnvName); ok {
+		return val
+	}
+	return kanisterToolsImage
+}
+
+func createRepositoryServerTestPod(ctx context.Context, cli kubernetes.Interface, namespace string, secret *corev1.Secret) (*corev1.Pod, error) {
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: repositoryServerTestPod,
+			Namespace:    namespace,
+			Labels:       map[string]string{"app": "test-repo-server"},
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name:  "test-repository-server-container",
+					Image: getKanisterToolsImage(),
+					Ports: []corev1.ContainerPort{
+						{
+							HostPort:      51515,
+							ContainerPort: 51515,
+						},
+					},
+					Env: []corev1.EnvVar{
+						{
+							Name:  awsconfig.AccessKeyID,
+							Value: testAwsAccessKeyId,
+						},
+						{
+							Name:  awsconfig.SecretAccessKey,
+							Value: testAwsAccessSecretKey,
+						},
+						{
+							Name:  awsconfig.Region,
+							Value: testAwsRegion,
+						},
+						{
+							Name:  "LOCATION_ENDPOINT",
+							Value: testAwsLocationEndpoint,
+						},
+					},
+					VolumeMounts: []corev1.VolumeMount{
+						{
+							Name:      "tls-certs",
+							MountPath: "/tmp/tls/",
+							ReadOnly:  true,
+						},
+					},
+				},
+			},
+			Volumes: []corev1.Volume{
+				{
+					Name: "tls-certs",
+					VolumeSource: corev1.VolumeSource{
+						Secret: &corev1.SecretVolumeSource{
+							SecretName: secret.GetName(),
+						},
+					},
+				},
+			},
+		},
+	}
+
+	podCreated, err := cli.CoreV1().Pods(namespace).Create(ctx, pod, metav1.CreateOptions{})
+	if err != nil {
+		return nil, err
+	}
+	return podCreated, nil
+}
+
+func createRepositoryServerTestService(ctx context.Context, cli kubernetes.Interface, namespace string) (*corev1.Service, error) {
+	name := fmt.Sprintf("%s%s", repositoryServerTestService, rand.String(5))
+
+	service := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Labels:    map[string]string{"name": name},
+		},
+		Spec: corev1.ServiceSpec{
+			Type:     corev1.ServiceTypeNodePort,
+			Selector: map[string]string{"app": "test-repo-server"},
+			Ports: []corev1.ServicePort{
+				{
+					Name:       "test-repo-server-service-port",
+					Protocol:   "TCP",
+					Port:       51515,
+					NodePort:   31325,
+					TargetPort: intstr.FromInt(51515),
+				},
+			},
+		},
+	}
+
+	serviceCreated, err := cli.CoreV1().Services(namespace).Create(ctx, service, metav1.CreateOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	return serviceCreated, nil
+}
+
+func createTestS3CredsLocationSecrets(ctx context.Context, cli kubernetes.Interface, namespace string) (*corev1.Secret, *corev1.Secret, error) {
```

Comment:
There are utilities exposed in `pkg/testutils` for secret creation. Can we reuse them

Reply:
Done !